src/component/addnote.js

```diff
@@ -1,88 +1,85 @@
 import "../styles.css";
 import { useNotes } from "../context/noteProvider";
-import { v4 as uuid } from "uuid";
-import {Link} from "react-router-dom";
+// import {Link} from "react-router-dom";
+import dayjs from "dayjs";
+import { BsPin } from "react-icons/bs";
+import { MdPushPin } from "react-icons/md";
+import { Toasters } from "./toasters";
 
 export default function AddNote() {
-  const { state ,dispatch } = useNotes();
+  const { state ,dispatch, addNewNoteHandler } = useNotes();
 
   const labels = ["Home" , "Work" , "Study" , "Gym" , "Health"];
   const priority = ["High" , "Medium" , "Low"]
+  const colors = ["Blue" ,"Green" ,"Yellow", "Purple", "Orange"];
 
   return (
     <div className="AddNote">
-      <div className="add-note-container">
+      <form className="add-note-container" 
+      onSubmit={ (e) => {
+      e.preventDefault()
+      dispatch({ type: "SET_DATE", payload: dayjs().format("YYYY-MM-DD") })
+      addNewNoteHandler(state)
+      dispatch({ type: "CLEAR_NOTE" });
+      }}>
         <section className="note-title-container">
-          <textarea id="w3review" name="notetittle" rows="2" cols="50" placeholder="Note Tittle" className="note-textarea" value={state.title}
-            onChange={(e) => dispatch({
-              type: "SET_NOTE",
-              value: e.target.value,
-              key: "title"
-            })}/>
+          <textarea id="w3review" name="notetittle" rows="2" cols="50" placeholder="Note Tittle" className="note-textarea" 
+          value={state.title}
+          onChange={(e) => dispatch({ type: "SET_TITLE", payload: e.target.value })}/>
         </section>
         <section className="note-body-container">
-          <textarea id="w3review" name="notebody" rows="6" cols="50" placeholder="Note body" className="note-textarea" value={state.body}
-            onChange={(e) => dispatch({
-              type : "SET_NOTE",
-              value : e.target.value,
-              key : "body"
-            })}/>
+          <textarea id="w3review" name="notebody" rows="6" cols="50" placeholder="Note body" className="note-textarea" value={state.content}
+                    onChange={(e) => dispatch({ type: "SET_CONTENT", payload: e.target.value })}/>
         </section>
+        <ul className="note-label-container list p1 bold">
+          {state.pin ?<li><MdPushPin name="pin item" className="p3 text-grey ptr" title="pinned note" onClick={() => {dispatch({ type: "SET_PIN", payload: !state.pin })}}/><p>Unpin Note</p></li> 
+          : <li><BsPin name="pin item" className="p3 text-grey ptr" title="unpined note" onClick={() => {dispatch({ type: "SET_PIN", payload: !state.pin })}}/><p>Pin Note</p></li> }  
```

Comment:
Suggestion: can fix code indentation here so it get easy to read to other person